components/base-node.js

```diff
@@ -44,6 +46,18 @@ module.exports = {
    },
 
   /**
+   * Standard component error handling - send a message to the error port and throw Error
+   *
+   * @param message message to be sent to error port
+   * @param errorPort handle to the component's error port
+   */
+   handleError: function( message, errorPort ) {
```

Comment:
It is up to the user (e.g, Alice), whether they choose to hook up the error port.  They don't have to.  If they do not, the clutter is minimal.    I have started to centralize my error handling in that new baseNode.handleError API - should make it easy to add logging there.